dspy/clients/self_hosted.py

```diff
@@ -0,0 +1,34 @@
+from typing import Dict, Any
+
+from dspy.clients.openai import is_openai_model
+
+
+#-------------------------------------------------------------------------------
```

Comment:
This file should be a class, that inherits from some ABC that defines the right interface. Right now, we're essentially expressing a class via 3 disconnected functions.

Reply:
I would rather just delete this file until we have real code logic

Reply:
(Looks like deleted)